code/modules/mob/mob.dm

```diff
@@ -1773,5 +1777,48 @@ mob/proc/on_foot()
 	ear_deaf = init_deaf
 	clear_fullscreen("blind")
 
+/mob/proc/transmogrify(var/target_type, var/offer_revert_spell = FALSE)
+	if(!target_type)
+		if(transmogged_from)
+			transmogged_from.forceMove(loc)
+			if(key)
+				transmogged_from.key = key
+			transmogged_from.timestopped = 0
+			if(istype(transmogged_from, /mob/living/carbon))
+				var/mob/living/carbon/C = transmogged_from
+				if(C.wear_mask && istype(C.wear_mask, /obj/item/clothing/mask/morphing))
+					C.drop_item(C.wear_mask, force_drop = 1)
+			transmogged_from = null
+			for(var/atom/movable/AM in contents)
+				AM.forceMove(get_turf(src))
+			qdel(src)
+		return
+	var/mob/M = new target_type(loc)
+	if(!istype(M))
```

Comment:
You should probably validate whether the path is a mob beforehand instead.
`ispath(target_type, /mob)`

And make sure to throw an exception when it isn't.